### ISSUE TITLE
When Testcontainers is selected, add an entry to HELP.md about Docker image tags and the need to update them

### ISSUE THREAD

**Issue by sivaprasadreddy** [spring-io/start.spring.io#1316]
Currently, when we select **Testcontainers** or **Docker Compose** starter and add a database like **PostgreSQL**, the image tag "latest" is used.

```
@TestConfiguration(proxyBeanMethods = false)
public class TestMyappApplication {

	@Bean
	@ServiceConnection
	PostgreSQLContainer<?> postgresContainer() {
		return new PostgreSQLContainer<>(DockerImageName.parse("postgres:latest"));
	}

	public static void main(String[] args) {
		SpringApplication.from(MyappApplication::main).with(TestMyappApplication.class).run(args);
	}

}
```

Generally, using **"latest"** tag is not a recommended practice, and we should be using the specific version that we use in production.

I know it is the developer's responsibility to change it to the version they should be using.
Still, I think it would be better to use a recent version than using "latest" to promote good practices by default. 

I have explored where the tag is specified and found that it is in https://github.com/spring-io/start.spring.io/blob/main/start-site/src/main/java/io/spring/start/site/container/SimpleDockerServiceResolver.java.

Can we define container versions as constants in SimpleDockerServiceResolver.java and use them?

**Comment by wilkinsona** [comment]
Thanks for the suggestion.

We've already considered doing this but realised that there's no way for us to pick the right version for each user's needs. The versions that you use in both Testcontainers configuration and Docker Compose configuration should match the versions of the services that you're using in production and we have no way of knowing what they are.

We decided that the least bad option was to use `latest` and to add a note to the generated `HELP.md` file that recommends changing the version. For example:

> This project contains a Docker Compose file named compose.yaml. In this file, the following services have been defined:
>
> - postgres: [postgres:latest](https://hub.docker.com/_/postgres)
>
> Please review the tags of the used images and set them to the same as you're running in production.

Thanks to this issue, I've realised that we've only done this for Docker Compose and not for Testcontainers. I think we should correct that and generate a similar section in the HELP.md when Testcontainers is selected.
